userspace/libscap/scap_savefile.c

```diff
@@ -2523,6 +2748,70 @@ int32_t scap_next_offline(scap_t *handle, OUT scap_evt **pevent, OUT uint16_t *p
 		*pevent = (struct ppm_evt_hdr *)(handle->m_file_evt_buf + sizeof(uint16_t));
 	}
 
+	if((*pevent)->type >= PPM_EVENT_MAX)
+	{
+		//
+		// We're reading a capture that contains new syscalls.
+		// We can't do anything else that skips them.
+		//
+		return scap_next_offline(handle, pevent, pcpuid);

```

Comment:
While this is technically correct, I wonder if we can perhaps use an iterative version to protect ourselves from a long sequence of new events causing a stack overflow?

Reply:
Yes, will do.